apps/openmw/mwmechanics/spells.hpp

```diff
@@ -32,37 +30,35 @@ namespace MWMechanics
     /// diseases. It also keeps track of used powers (which can only be used every 24h).
     class Spells
     {
-        public:
-
-            typedef const ESM::Spell* SpellKey;
-            struct SpellParams
-            {
-                std::map<int, float> mEffectRands; // <effect index, normalised random magnitude>
-                std::set<int> mPurgedEffects; // indices of purged effects
-            };
-
-            typedef std::map<SpellKey, SpellParams> TContainer;
-            typedef TContainer::const_iterator TIterator;
-
-        private:
-            TContainer mSpells;
+            std::shared_ptr<SpellList> mSpellList;
+            std::map<const ESM::Spell*, SpellParams> mSpells;
```

Comment:
A pretty intrusive change. Make sure that you import data from older saves and write it to new saves properly.

Reply:
I was able to load an older save, resave it, and load it again.